staging/src/k8s.io/api/resource/v1beta2/types.go

```diff
@@ -804,7 +804,9 @@ type DeviceSubRequest struct {
 	// +optional
 	AllocationMode DeviceAllocationMode `json:"allocationMode,omitempty" protobuf:"bytes,4,opt,name=allocationMode"`
 
-	// Count is used only when the count mode is "ExactCount". Must be greater than zero.
+	// Count is used only when the count mode is "ExactCount". Must be greater than zero
+	// unless the subrequest is the last entry in the FirstAvailable list. For the last entry,
+	// a count of zero is allowed.
```

Comment:
Is it valid to have a single entry with a zero count?

Reply:
Good point. This PR initially allowed this, but I don't think we should, so I've updated it to only allow `Count: 0` for the last entry if there are more than one entry in the `FirstAvailable` list.

The drawback here is that the rule for what is allowed becomes a bit more complicated. But we don't allow `Count: 0` for regular requests (without subrequests), so it would be strange to allow this for a single subrequest, which is essentially identical to just having a normal request.

@pohly